src/semver/version.py

```diff
@@ -68,15 +68,19 @@ class Version:
     __slots__ = ("_major", "_minor", "_patch", "_prerelease", "_build")
     #: Regex for number in a prerelease
     _LAST_NUMBER = re.compile(r"(?:[^\d]*(\d+)[^\d]*)+")
-    #: Regex for a semver version
-    _REGEX = re.compile(
+    #: Regex template for a semver version
+    _REGEX_TEMPLATE = \
         r"""
             ^
             (?P<major>0|[1-9]\d*)
-            \.
-            (?P<minor>0|[1-9]\d*)
-            \.
-            (?P<patch>0|[1-9]\d*)
+            (?:
+                \.
+                (?P<minor>0|[1-9]\d*)
+                (?:
+                    \.
+                    (?P<patch>0|[1-9]\d*)
+                ){}
+            ){}
```

Comment:
To make it a bit more explicit, I would introduce some placeholders:

```suggestion
                ){opt_patch}
            ){opt_minor}
```

Reply:
Yes, great idea 👍